src/App.tsx

```diff
@@ -11,29 +12,73 @@ import { OnboardContext } from "common/context/OnboardContext";
 import { recoverPublicKey } from "./features/vote/helpers/recoverPublicKey";
 import { derivePrivateKey } from "./features/vote/helpers/derivePrivateKey";
 
+import currentSigningMessage from "common/currentSigningMessage";
+import { useCurrentRoundId } from "hooks";
+
 interface Props {
   queryClient: QueryClient;
 }
 
-export interface SigningKeys {
+export interface SigningKey {
   [key: string]: {
     publicKey: string;
     privateKey: string;
   };
 }
 
+export interface SigningKeys {
+  [key: string]: SigningKey;
+}
+
+const SIGNING_KEYS_STORAGE_KEY = "signingKeys";
+
 function App(props: Props) {
   const [signingKeys, setSigningKeys] = useState<SigningKeys>({});
 
   const { state, disconnect, dispatch } = useContext(OnboardContext);
   const { error, removeError, addError } = useContext(ErrorContext);
+  const { data: currentRoundId } = useCurrentRoundId();
 
   useEffect(() => {
     if (state.signer && state.address) {
       const address = state.address;
-      const message = "Login to UMA Voter dApp";
-      const keyExists = signingKeys[address];
-      if (!keyExists) {
+      const message = currentSigningMessage(Number(currentRoundId));
+      const hashedMessage = ethers.utils.formatBytes32String(message);
+      const keysString = localStorage.getItem(SIGNING_KEYS_STORAGE_KEY);
+      if (keysString) {
+        const keys = JSON.parse(keysString) as SigningKeys;
+        const keyExists = keys[hashedMessage][address];
+        if (!keyExists) {
+          state.signer
+            .signMessage(message)
+            .then((msg) => {
+              const key = {} as { publicKey: string; privateKey: string };
```

Comment:
Is this code block the same as below? Could we make this a separate function or somehow reuse this code so it's not repeated?